### PR TITLE
Added custom message option to cast_assoc and cast_embed

### PR DESCRIPTION
**Pull request by acarpe**

### CODE REVIEW

**Comment by josevalim** [comment]
Thank you @acarpe. Because we have both "can't be blank" and "is invalid" messages, what if we support `:on_blank` and `:on_invalid` parameters for messages? @michalmuskala thoughts on how the options should be named considering our existing API?

**Comment by michalmuskala** [comment]
How about `blank_message` and `invalid_message`? I think it fits better into other functions where we use `message` for the validation information.

**Comment by josevalim** [comment]
Let's go with `required_message` and `invalid_message`. That would also mirror the `validate_required` bit. @acarpe can you please amend your commit accordingly? Thank you!

**Comment by acarpe** [comment]
Changes done, you can review it :)

File: lib/ecto/changeset.ex [hunk 1]
```diff
@@ -511,10 +515,10 @@ defmodule Ecto.Changeset do
     end
   end
 
-  defp missing_relation(%{changes: changes, errors: errors} = changeset, name, current, required?, relation) do
+  defp missing_relation(%{changes: changes, errors: errors} = changeset, name, current, required?, relation, opts) do
     current_changes = Map.get(changes, name, current)
     if required? and Relation.empty?(relation, current_changes) do
-      %{changeset | errors: [{name, {"can't be blank", []}} | errors], valid?: false}
+      %{changeset | errors: [{name, {(opts[:required_message] || "can't be blank"), []}} | errors], valid?: false}
```
Comment (michalmuskala) [inline lib/ecto/changeset.ex]: Should we change `message/2` to accept additional message key?
````
defp message(opts, key \\ :mesage, default)
```

**Comment by acarpe** [comment]
@michalmuskala modified the message function.

**Comment by michalmuskala** [comment]
:shipit:

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**